minahilfaisal/frontend-app-authn
# Organization Field - Added to Registration Form

## Description

This pull request adds a new "Organization" field to the registration form for the `frontend-app-authn` MFE, allowing users to specify their affiliated organization during the registration process.

This dropdown field lists the organizations available in the `lms` database, while also allowing users to add in an organization that is not available in the list.

## Changes Made

- Added an input field for the "Organization" information in the registration form.
- Ensured that proper styling was applied to the new field added.
- Validated the input at the frontend to ensure the field the shows input error. The message for the field being required is also shown in case of no input.
- Wrote redux code to receive data from API (list of organizations available) and store it in state.
- Wrote tests for the new field and modified the other tests to include the Organization field when registering a new user.

## How to Test

1. Setup [backend PR](https://github.com/minahilfaisal/edx-platform/pull/1) first.
2. Access the registration form on the [website](http://localhost:1999/register).
3. Observe the presence of the "Organization" field in the form.
4. Enter valid organization information or pick an organization from the dropdown and complete the registration process.
5. Check the user profile from the [admin page](http://localhost:18000/admin/auth/user/) and see that the organization field value has been correctly stored.

## Issues / Improvements

1. An error occurs when submitting the form. According to further investigation I have come to conclusion that it occurs on the Master branch as well. I have been unable to resolve it.
Stack trace is as follows in the screenshot:
<img width="512" alt="Screenshot 2023-11-02 at 11 29 18 AM" src="https://github.com/minahilfaisal/frontend-app-authn/assets/141846591/c4f61d45-7803-408b-afef-87a7d85d4ae7">
2. 49 out of 50 tests written in the `src/register/RegistrationPage.test.jsx` file pass. I am unable to resolve this issue or fix the test.

The relevant tests can be run using the following command: 
`npm test -- RegistrationPage.test.jsx`

## Screenshots

Screenshot of MFE showing organization field in the registration form.
<img width="512" alt="Screenshot 2023-11-05 at 6 54 47 PM" src="https://github.com/minahilfaisal/frontend-app-authn/assets/141846591/09aeaeb5-43f3-4a5f-95cb-a01067168e18">

Screenshot showing dropdown values for the Organization field.
<img width="488" alt="Screenshot 2023-11-05 at 6 55 14 PM" src="https://github.com/minahilfaisal/frontend-app-authn/assets/141846591/f5a5df67-5cf5-49cc-af1f-b441b405d17d">

Screenshot of admin page after successful registration of user with Demo Organization.
<img width="683" alt="Screenshot 2023-11-05 at 6 51 31 PM" src="https://github.com/minahilfaisal/frontend-app-authn/assets/141846591/eb84e14f-eda7-4144-8b10-e4746357e2ad">

## Checklist

- [ ] The code follows our established coding guidelines.
- [ ] Unit tests have been added to ensure the proper functionality of the new field.
- [ ] Documentation has been updated to reflect the changes made.

Please review and provide feedback. Thank you.
